### PR TITLE
generic/tests/os_update: refactor and have MultiVM support

### PR DESCRIPTION
**Pull request by balamuruhans**
yum_update.py had support to perform os update for RHEL based guests,
refactor it to support Ubuntu and adopt MultiVM. The patch also renames
yum_update.py -> os_update.py similarly cfg file is renamed.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: generic/tests/os_update.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vms = env.get_all_vms()
+    timeout = int(params.get("os_update_timeout"))
+    for vm in vms:
+        if not vm.is_alive():
+            vm.start()
+        cmd = "yum update -y && yum update kernel -y"
```
Comment (luckyh) [inline generic/tests/os_update.py]: I think that only having `"yum update -y"` is ok.

**Comment by luckyh** [comment]
@ldoktor @yashkmankad I'm not sure if this change would affect your test, so please help review it as well, thanks.

File: generic/tests/os_update.py (hunk 1 above)
Comment (balamuruhans) [reply generic/tests/os_update.py]: sure, I tried to keep not to change the existing one. I can do that.

**Comment by ldoktor** [comment]
Hello guys, what a nice improvement. I'd rather see the use of `avocado.utils.package_manager`, but I only now noticed we haven't implemented the RemoteCommander there yet. Card about it here: https://trello.com/c/dzYRnW98/1401-add-processrun-support-to-softwaremanager

**Comment by ldoktor** [comment]
Forgot to mention I haven't thoroughly reviewed this patch, changing the name of this test could bring some problems to current users, but in the long run it should be more consistent. So conceptually yes from me and hope users will find the new test name :smile:

File: generic/tests/cfg/os_update.cfg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- os_update: install setup image_copy unattended_install.cdrom
+    virt_test_type = qemu libvirt
```
Comment (sathnaga) [inline generic/tests/cfg/os_update.cfg]: we can mark `no Windows and Sles` explicitly

Comment (sathnaga) [inline generic/tests/cfg/os_update.cfg]: we can mark `no Windows and Sles` explicitly

**Comment by balamuruhans** [comment]
@luckyh @vivianQizhu can we consider this patch for merge

**Comment by luckyh** [comment]
Sorry for the noises, github just backed from an incident. Let's get this in.

**Comment by balamuruhans** [comment]
Thank you all.

**Pull request merged**